R/deps.R

```diff
@@ -280,25 +280,39 @@ update.package_deps <- function(object, ..., quiet = FALSE, upgrade = TRUE) {
 
 install_packages <- function(packages, repos = getOption("repos"),
                              type = getOption("pkgType"), ...,
-                             dependencies = FALSE, quiet = NULL,
-                             # These are options to `install()` used when
-                             # installing remotes, but can get passed to us by
-                             # `...` so we just ignore them here
-                             build, build_opts) {
+                             dependencies = FALSE, quiet = NULL) {
+
+  # We want to pass only args that exist in the downstream functions
```

Comment:
I don't really love this, but I am not sure of a better way to do it, you could list even more arguments here to catch them, but that seems like it would be a pain to maintain as the possible number of arguments in the `install_()` functions increases.

Reply:
Could we required that all install functions take `...` and then use ellipsis to warn if an argument is never evaluated?

Reply:
remotes does not have dependencies, by design. ellipsis already has compiled code, and will trigger the DLL locking issue on Windows.

Reply:
Oh, oops, I forgot about that constraint.